libminifi/include/FlowFileRepository.h

```diff
@@ -0,0 +1,208 @@
+/**
+ * @file FlowFileRepository 
+ * Flow file repository class declaration
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef __FLOWFILE_REPOSITORY_H__
+#define __FLOWFILE_REPOSITORY_H__
+
+#include <ftw.h>
+#include <uuid/uuid.h>
+#include <atomic>
+#include <cstdint>
+#include <cstring>
+#include <iostream>
+#include <map>
+#include <set>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "Configure.h"
+#include "Connection.h"
+#include "FlowFileRecord.h"
+#include "Logger.h"
+#include "Property.h"
+#include "ResourceClaim.h"
+#include "io/Serializable.h"
+#include "utils/TimeUtil.h"
+#include "Repository.h"
+
+class FlowFileRepository;
+
+//! FlowFile Event Record
+class FlowFileEventRecord : protected Serializable
+{
+public:
+	friend class ProcessSession;
+public:
+	//! Constructor
+	/*!
+	 * Create a new provenance event record
+	 */
+	FlowFileEventRecord()
+	: _entryDate(0), _lineageStartDate(0), _size(0), _offset(0)  
+	{
+		_eventTime = getTimeMillis();
+		logger_ = Logger::getLogger();
+	}
+
+	//! Destructor
+	virtual ~FlowFileEventRecord() {
+	}
+	//! Get Attributes
+	std::map<std::string, std::string> getAttributes() {
```

Comment:
what do you mean somehow? can you explain a path? Why would you be worrying about an event record being deleted? If that is the case we have bigger problems. perhaps we should discuss this.